pkg/controller/registry/resolver/rbac.go

```diff
@@ -2,19 +2,38 @@ package resolver
 
 import (
 	"fmt"
+	"hash/fnv"
 
 	corev1 "k8s.io/api/core/v1"
 	rbacv1 "k8s.io/api/rbac/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apiserver/pkg/storage/names"
+	utilrand "k8s.io/apimachinery/pkg/util/rand"
 
 	"github.com/operator-framework/api/pkg/operators/v1alpha1"
 	"github.com/operator-framework/operator-lifecycle-manager/pkg/controller/install"
+	hashutil "github.com/operator-framework/operator-lifecycle-manager/pkg/lib/kubernetes/pkg/util/hash"
 	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/ownerutil"
 )
 
-var generateName = func(base string) string {
-	return names.SimpleNameGenerator.GenerateName(base + "-")
+const (
+	maxNameLength          = 63
+	randomLength           = 5
+	maxGeneratedNameLength = maxNameLength - randomLength
+)
+
+// HashObject calculates a hash of the object for stable name generation
+func HashObject(o interface{}) string {
+	hasher := fnv.New32a()
+	hashutil.DeepHashObject(hasher, o)
+	return utilrand.SafeEncodeString(fmt.Sprint(hasher.Sum32()))
+}
+
+func generateName(base string, hash string) string {
```

Comment:
Is the contract of this function to return the concatenation of base+hash, truncating base such that the result is short enough to be an object name? I'm worried about slice range out-of-bound panics and overly-long return values. Couldn't base have length 127 in the invocation on L116?

Reply:
pushed tests for this - thanks for calling it out. see `rbac_test.go`